src/traces/table/plot.js

```diff
@@ -0,0 +1,892 @@
+                var rowLengthLimit = d.column.columnWidth - 2 * c.cellPad;
+                d.value = '';
+                while(rest.length) {
+                    currentAddition = rest.shift();
+                    currentAdditionLength = currentAddition.width + separatorLength;
+                    if(currentRowLength + currentAdditionLength > rowLengthLimit) {
+                        d.value += currentRow.join(c.wrapSpacer) + c.lineBreaker;
+                        currentRow = [];
+                        currentRowLength = 0;
+                    }
+                    currentRow.push(currentAddition.text);
+                    currentRowLength += currentAdditionLength;
+                }
+                if(currentRowLength) {
+                    d.value += currentRow.join(c.wrapSpacer);
+                }
+                d.wrapped = true;
+            });
+
+        // the pre-wrapped text was rendered only for the text measurements
+        cellTextHolder.selectAll('tspan.line').remove();
+
+        // resupply text, now wrapped
+        populateCellText(cellTextHolder.select('.cellText'), tableControlView, columnBlock);
+        d3.select(element.parentNode.parentNode).call(setCellHeightAndPositionY);
+    };
+}
+
+function updateYPositionMaker(columnBlock, element, tableControlView, gd, d) {
+    return function updateYPosition() {
+        if(d.settledY) return;
+        var cellTextHolder = d3.select(element.parentNode);
+        var l = getBlock(d);
+        var rowIndex = d.key - l.firstRowIndex;
+
+        var declaredRowHeight = l.rows[rowIndex].rowHeight;
+
+        var requiredHeight = d.cellHeightMayIncrease ? element.parentNode.getBoundingClientRect().height + 2 * c.cellPad : declaredRowHeight;
+
+        var finalHeight = Math.max(requiredHeight, declaredRowHeight);
+        var increase = finalHeight - l.rows[rowIndex].rowHeight;
+
+        if(increase) {
+
+            // current row height increased
+            l.rows[rowIndex].rowHeight = finalHeight;
+
+            columnBlock
+                .selectAll('.columnCell')
+                .call(setCellHeightAndPositionY);
+
+            updateBlockYPosition(null, columnBlock.filter(cellsBlock), 0);
+
+            // if d.column.type === 'header', then the scrollbar has to be pushed downward to the scrollable area
+            // if d.column.type === 'cells', it can still be relevant if total scrolling content height is less than the
+            //                               scrollable window, as increases to row heights may need scrollbar updates
+            renderScrollbarKit(tableControlView, gd);
+        }
+
+        cellTextHolder
+            .attr('transform', function() {
+                // this code block is only invoked for items where d.cellHeightMayIncrease is truthy
+                var element = this;
+                var columnCellElement = element.parentNode;
+                var box = columnCellElement.getBoundingClientRect();
```

Comment:
Can we avoid these `getBoundingClientRect` calls?

Reply:
The key test here (which I assume will fail given the `getBoundingClientRect` call) is to call `Plotly.newPlot` into an element that's not in the DOM, and insert it in the DOM afterward - ala https://codepen.io/anon/pen/XjakQy
If you really *do* need a `getBoundingClientRect` result, use `Drawing.bBox`.

Reply:
I think that https://github.com/plotly/plotly.js/issues/2053 should address it. I added a 'core' line now to express this. Here, the difference between parent and child bounding box top matters. What do people, or we, do with detached nodes? I'm trying to minimize open points for this specific PR, unless of course users would run into serious issues with it otherwise.